sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -923,6 +922,13 @@ def test_asin_series():
     assert t5 == 3*x**5/40
     assert asin(x).taylor_term(7, x, t5, 0) == 5*x**7/112
 
+def test_asin_leading_term():
+    assert asin(x).as_leading_term(x) == x
+    assert asin(x + 1).as_leading_term(x) == pi/2
+    assert asin(x - 1).as_leading_term(x) == -pi/2
+    assert asin(1/x).as_leading_term(x, cdir=1).expand() == I*log(x) + pi/2 - I*log(2)
```

Comment:
It is probably better to add `expand` to `asin._eval_as_leading_term` (and to other similar methods where `rewrite(log)` will introduce nontrivial factors).